p2p_wallet/Scenes/CreateOrRestoreWallet/Common/Pincode/Pincode/PincodeViewModel.swift

```diff
@@ -7,18 +8,26 @@ import UIKit
 enum PincodeState {
     case create
     case confirm(pin: String, askBiometric: Bool)
+    case check
 }
 
 final class PincodeViewModel: BaseViewModel {
     // MARK: - Dependencies
 
     @Injected private var biometricsAuthProvider: BiometricsAuthProvider
     @Injected private var authenticationHandler: AuthenticationHandlerType
+    @Injected private var pincodeService: PincodeService
+    @Injected private var userWalletManager: UserWalletManager
+    @Injected private var notificationService: NotificationService
 
     // MARK: - Properties
 
     @Published var title: String = ""
     @Published var snackbar: PincodeSnackbar?
+    @Published var showAlert: (String, String)?
+    @Published var showForgetPin: Bool = false
+    @Published var showFaceid: Bool = false
```

Comment:
do we need to show touch id?

Reply:
yes, in case of check and if it's available

Reply:
so it is only about faceid now?

Reply:
> so it is only about faceid now?
I mean, we use faceid icon for biometry no matter what kind of it is (faceid or touchid)

Reply:
oh ok